rust/lance/src/datafusion/dataframe.rs

```diff
@@ -96,7 +96,9 @@ impl TableProvider for LanceTableProvider {
                     columns.push(self.full_schema.field(*field_idx).name());
                 }
             }
-            if !columns.is_empty() {
+            if projection.is_empty() {
```

Comment:
this condition can happen before 

```
let mut columns = Vec::with_capacity(projection.len());
            for field_idx in projection {
                if Some(*field_idx) == self.row_id_idx {
                    scan.with_row_id();
                } else if Some(*field_idx) == self.row_addr_idx {
                    scan.with_row_address();
                } else {
                    columns.push(self.full_schema.field(*field_idx).name());
                }
            }
```

to avoid initializing the columns and computing all the logic above.